napari/components/_tests/test_viewer_mouse_bindings.py

```diff
@@ -38,12 +38,12 @@ def inverted(self):
         (
             ["Control"],
             WheelEvent(False),
-            [[5, 5, 5], [5, 5, 4], [5, 5, 3], [5, 5, 0]],
+            [[5, 5, 5], [4, 5, 5], [3, 5, 5], [0, 5, 5]],
```

Comment:
You are correct! Checked locally and indeed the `last_used` value fallbacks to 0 after setting it to 2 in the test. Also, the addition of a warning makes sense to me :+1: it's true that is not so trivial to understand why, after `last_used` is explicitly set, it can inmediatly have another value (could even make users think there is an issue if not informed about the validation and fallback to a valid value)

Reply:
In that case, I think we should at least update the test with `viewer.dims.last_used = 0`.

I don't feel strongly about the warning, especially because the `Dims` root validator does a lot of this type of coercion without warning.

Reply:
I don't have much intuition regarding this, sorry.
On the GUI side, the behavior seems logical to me (other than clicking on the roll dims button stealing keyboard focus from sliders, but that's a different issue/PR).
The bugfix code makes sense to me, so I think this is OK and we can revisit if needed?